lib/fusion_auth/openid_connect.ex

```diff
@@ -0,0 +1,223 @@
+    |> FusionAuth.result()
+  end
+
+  @doc """
+  Completes an OpenID connect login using a login request, this function allows for optional headers to be passed. *Must* at least contain the following values
+  - applicationId :: String.t()\n
+    The Id of the Application the user is to be logged into.
+
+  - data
+    - code :: String.t()\n
+    The `code` parameter that was returned to the Authorization redirect URI.
+
+    -redirect_uri :: String.t()\n
+    The redirect URI that was provided to the OpenID Connect Authorization endpoint
+
+  - identityProviderId :: String.t()\n
+    The Id of the identity provider to process this login request.
+
+  For more information, visit the FusionAuth API documentation for [Complete an OpenID connect login](https://fusionauth.io/docs/v1/tech/apis/identity-providers/openid-connect#complete-an-openid-connect-login)
+  """
+  @spec complete_openid_connect_login(client(),login_request_body()) :: FusionAuth.result()
+  def complete_openid_connect_login(client, login_request_body) do
+    Tesla.post(client, @connect_url <> "/login", login_request_body)
+    |> FusionAuth.result()
+  end
+
+  @doc """
+  Completes an OpenID connect login using a login request, this function allows for optional headers to be passed. *Must* at least contain the following values
+  - applicationId :: String.t()\n
+    The Id of the Application the user is to be logged into.
+
+  - data
+    - code :: String.t()\n
+    The `code` parameter that was returned to the Authorization redirect URI.
+
+    -redirect_uri :: String.t()\n
+    The redirect URI that was provided to the OpenID Connect Authorization endpoint
+
+  - identityProviderId :: String.t()\n
+    The Id of the identity provider to process this login request.
+
+  - headers :: list()\n
+    list of headers in the form of tuples
+
+  - header :: tuple()\n
+    header in the form of {"header-name", "header-value"}
+
+  Headers available\n
+    - `X-Forwarded-For`
+      The IP address of a client requesting authentication. If the IP address is provided it will be
+      stored in the user’s login record. It is generally preferred to specify the IP address in the
+      request body. If it is not provided in the request body this header value will be used if available.
+      However, the request body value takes precedence.
+    - `X-FusionAuth-TenantId`
+      The unique Id of the tenant used to scope this API request.
+
+  For more information, visit the FusionAuth API documentation for [Complete an OpenID connect login](https://fusionauth.io/docs/v1/tech/apis/identity-providers/openid-connect#complete-an-openid-connect-login)
+  """
+  @spec complete_openid_connect_login(client(),login_request_body(), headers()) :: FusionAuth.result()
+  def complete_openid_connect_login(client, login_request_body, headers) do
+    Tesla.post(client, @connect_url <> "/login", login_request_body, headers: headers)
+    |> FusionAuth.result()
+  end
+
```

Comment:
Let's just cleanup the whitespace at the end of the file. We _could_ probably add a `.vscode` folder with proper formatting setting to this repo if wanted. Just a note not an action item or anything